packages/storefront-sdk/src/client/index.ts

```diff
@@ -1,15 +1,77 @@
 import { createClient } from '@urql/core';
 import type { Client as UrqlClient } from '@urql/core';
 import type { StorefrontClientParams } from '../index.js';
+import type {
+	SetConfigParams,
+	SetConfigResponse,
+	StorefrontConfig
+} from '../types/config.js';
 
 export class StorefrontClient {
-	readonly #graphqlClient: UrqlClient;
+	#graphqlClient: UrqlClient;
+	#storefrontEndpoint: string;
+	#previewToken: string | undefined;
+	#currencyCode: string | undefined;
+	#locale: string | undefined;
```

Comment:
In `@nacelle/storefront-sdk@1.x`, we set the `locale` to `'en-US'` if `params.locale` is `undefined`.

I think we would be justified in not doing that (and leaving this type as `string | undefined`) in `@2.x`, because the Storefront GraphQL's `ContentFilterInput`, `ProductCollectionFilterInput`, and `ProductFilterInput` set the `locale` to `'en-US'` if a `locale` is not provided. TLDR: it's handled upstream, so we don't need to handle it here.